### PR TITLE
Adding Biography and Making it Editable

### PR DESCRIPTION
**Pull request by alisonrosenman**
Let me know how this looks — One small bug I can’t figure out is that when you go to update the bio, everything works fine (if you were to reload the bio page, the new bio would be there and everything!) but upon hitting the submit button you get to a blank page then have to hit the back button.

### CODE REVIEW

File: src/main/java/codeu/controller/ProfileServlet.java [hunk 1]
```diff
@@ -81,5 +81,10 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
   /** This function fires when a user submits the form on the profile page. */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response)
-      throws IOException, ServletException {}
+      throws IOException, ServletException {
+    String requestUrl = request.getRequestURI();
+    String name = requestUrl.substring("/profile/".length());
+    User user = userStore.getUser(name);
+    user.biography = request.getParameter("newBio");
+  }
```
Comment (kkane098) [inline src/main/java/codeu/controller/ProfileServlet.java]: I believe if you add response.sendRedirect(requestUrl); at the end you can avoid being sent to a blank page.

File: src/main/webapp/WEB-INF/view/profile.jsp [hunk 2]
```diff
@@ -47,6 +47,23 @@ User user = (User) request.getAttribute("user");
   </head>
   <body onload="scrollChat()">
   <h1 align ="center"><%= user.getName() %>'s Profile</h1>
+  <div id="container">
+  <h2>Biography</h2>
```
Comment (kkane098) [inline src/main/webapp/WEB-INF/view/profile.jsp]: This looks like it would allow anyone to edit anyone else's biography, and we probably don't want that.

Comment (alisonrosenman) [reply src/main/webapp/WEB-INF/view/profile.jsp]: Ah, so it’s on a unique user’s page but anyone will eventually be able to get there.  I will try to keep it about the same but then have a password check or something like that. Thanks!

Comment (kkane098) [reply src/main/webapp/WEB-INF/view/profile.jsp]: I would just only display the form if the current logged in user is equal to the user associated with the profile

Comment (alisonrosenman) [reply src/main/webapp/WEB-INF/view/profile.jsp]: Ok I think this is fixed thanks everyone for the help!

**Comment by Mariamcm90** [comment]
I saw your post on the other pull request and I think the changes look good to me! 

One thing that I noticed is that in the PersistentDataStore class in the writeThrough method for users, maybe you can add a line with "userEntity.setProperty("biography", user.getBio());" to add that property to the User entity (although I think the way we set it up, it's initially null). In the section above for loading the users we have a getProperty section but no matching setProperty for the writes. I think then later we can add a method for retrieving and updating the value of the property to save the bios.

File: src/main/java/codeu/model/data/User.java [hunk 3]
```diff
@@ -54,6 +56,14 @@ public String getPassword() {
     return hashedPassword;
   }
 
+  public String getBio() {
+    return biography;
+  }
+
+  public void changeBio(String newBio) {
```
Comment (fatlotus) [inline src/main/java/codeu/model/data/User.java]: nit: idiomatic Java prefers `setXXX(String)`

File: src/main/java/codeu/model/store/persistence/PersistentDataStore.java [hunk 4]
```diff
@@ -65,11 +65,12 @@ public PersistentDataStore() {
         UUID uuid = UUID.fromString((String) entity.getProperty("uuid"));
         String userName = (String) entity.getProperty("username");
         String password = (String) entity.getProperty("password");
+        String biography = (String) entity.getProperty("biography");  //I'm not sure what this Entity business is, hope this works?
```
Comment (fatlotus) [inline src/main/java/codeu/model/store/persistence/PersistentDataStore.java]: You're doing it right! :smile_cat:

File: src/main/webapp/WEB-INF/view/activityFeed.jsp [hunk 5]
```diff
@@ -24,6 +24,8 @@ List<Message> messages = (List<Message>) request.getAttribute("messages");
        <a>Hello <%= request.getSession().getAttribute("user") %>!</a>
        <a href="/activityFeed">Activity Feed</a>
        <a href="/conversations">Conversations</a>
+        <% String profileaddress = (String) "/profile/" + request.getSession().getAttribute("user"); %>  
+        <a href="<%=profileaddress %>">Your Profile</a>
```
Comment (fatlotus) [inline src/main/webapp/WEB-INF/view/activityFeed.jsp]: nit: Space after `<%=`; you could also simplify by doing `href="/profile/<%= request.getSession().getAttribute("user") %>"`

File: src/main/webapp/index.jsp [hunk 6]
```diff
@@ -13,8 +13,13 @@
   See the License for the specific language governing permissions and
   limitations under the License.
 --%>
+
+<%@ page import="codeu.model.data.User" %>
+
+
```
Comment (kkane098) [inline src/main/webapp/index.jsp]: ???

**Comment by alisonrosenman** [comment]
hooray!

**Pull request merged**